pkg/ring/model.go

```diff
@@ -117,15 +117,17 @@ func (d *Desc) FindIngestersByState(state IngesterState) []IngesterDesc {
 
 // Ready is true when all ingesters are active and healthy.
 func (d *Desc) Ready(heartbeatTimeout time.Duration) bool {
+	numTokens := len(d.Tokens)
 	for _, ingester := range d.Ingesters {
 		if time.Now().Sub(time.Unix(ingester.Timestamp, 0)) > heartbeatTimeout {
 			return false
 		} else if ingester.State != ACTIVE {
 			return false
 		}
+		numTokens += len(ingester.Tokens)
 	}
 
-	return len(d.Tokens) > 0
+	return numTokens > 0
```

Comment:
I find this change interesting - why do we want to go ready when we don't have any tokens?

Reply:
From slack:

I'm confused.  We don't.  The code says we're ready when we have >0 tokens.  Or am I misreading something?

Bryan Boreham [5:38 PM]
looks to me like you're adding the number of tokens from all ingesters

Tom Wilkie [5:38 PM]
Ah yeah we are - good catch
I see what I was trying to do now...
Its actually not a change - d.Tokens used to contain all the tokens (for all nodes)
not just our tokens (d is the ring "desc")

Bryan Boreham [5:40 PM]
oh, right

Tom Wilkie [5:40 PM]
Thats not to say the logic is correct
(its not _incorrect_ either - it means we won't go ready until someone has tokens, but we want to be more specific)

Bryan Boreham [5:42 PM]
Am I right in thinking the only use of the 'ready' state is to halt a rolling update if things aren't going well?

Tom Wilkie [5:42 PM]
Correct
its basically saying if the "ring" is ready, not the given ingester.
adding a check that we have tokens would be done in the lifecycler.

Reply:
#1112